apps/rush-lib/src/api/RushProjectConfiguration.ts

```diff
@@ -329,41 +328,60 @@ export class RushProjectConfiguration {
-              overlappingPathAnalyzer.addPathAndGetFirstEncounteredLabels(outputFolderName, operationName);
-            if (overlappingOperationNames) {
-              const overlapsWithOwnOperation: boolean = overlappingOperationNames?.includes(operationName);
-              if (overlapsWithOwnOperation) {
-                if (overlappingOperationNames.length === 1) {
-                  terminal.writeErrorLine(
-                    `Invalid "${RUSH_PROJECT_CONFIGURATION_FILE.projectRelativeFilePath}" ` +
-                      `for project "${project.packageName}". The project output folder name "${outputFolderName}" in ` +
-                      `operation ${operationName} overlaps with another folder name in the same operation.`
-                  );
-                } else {
-                  const otherOperationNames: string[] = overlappingOperationNames.filter(
-                    (overlappingOperationName) => overlappingOperationName !== operationName
-                  );
-                  terminal.writeErrorLine(
-                    `Invalid "${RUSH_PROJECT_CONFIGURATION_FILE.projectRelativeFilePath}" ` +
-                      `for project "${project.packageName}". The project output folder name "${outputFolderName}" in ` +
-                      `operation ${operationName} overlaps with other folder names in the same operation and with ` +
-                      `folder names in the following other operations: ${otherOperationNames.join(', ')}.`
-                  );
+      // For each phased command, check if any of its phases' output folders overlap.
+      for (const [, command] of repoCommandLineConfiguration.commands) {
+        if (command.commandKind === 'phased') {
+          const phasesOverlappingPathAnalyzer: OverlappingPathAnalyzer<string> =
+            new OverlappingPathAnalyzer<string>();
+
+          for (const operationName of command.phases) {
+            const operationSettings: IOperationSettings | undefined =
+              operationSettingsByOperationName.get(operationName);
+            if (operationSettings) {
+              if (operationSettings.outputFolderNames) {
+                for (const outputFolderName of operationSettings.outputFolderNames) {
+                  const overlappingOperationNames: string[] | undefined =
+                    phasesOverlappingPathAnalyzer.addPathAndGetFirstEncounteredLabels(
+                      outputFolderName,
+                      operationName
+                    );
+                  if (overlappingOperationNames) {
+                    const overlapsWithOwnOperation: boolean =
+                      overlappingOperationNames?.includes(operationName);
+                    if (overlapsWithOwnOperation) {
+                      if (overlappingOperationNames.length === 1) {
+                        terminal.writeErrorLine(
+                          `Invalid "${RUSH_PROJECT_CONFIGURATION_FILE.projectRelativeFilePath}" ` +
```

Comment:
Will this path be correct for config files inherited via `"extends"`?

Reply:
This always prints `"config/rush-project.json"`.